tests/foreman/api/test_setting.py

```diff
@@ -83,3 +88,81 @@ def test_positive_update_login_page_footer_text_with_long_string(self):
                 login.value = login_text
                 login = login.update(['value'])
                 self.assertEqual(login.value, login_text)
+
+    @stubbed()
+    @tier1
+    def test_negative_update_hostname_with_empty_fact(self):
+        """Update the Hostname_facts settings without any string(empty values)
+
+        :id: b8e260fc-e263-4292-aa2f-ab37085c7758
+
+        :expectedresults: Error should be raised on setting empty value for
+            hostname_facts setting
+
+        :caseautomation: notautomated
+        """
+
+    @tier1
+    def test_positive_update_hostname_prefix_without_value(self):
+        """Update the Hostname_prefix settings without any string(empty values)
+
+        :id: 3867488c-d955-47af-ac0d-71f4016391d1
+
+        :expectedresults: Hostname_prefix should be set without any text
+        """
+        hostname_prefix_id = [ele.id for ele in entities.Setting().search(
+            query={"per_page": 200, 'search': 'name="discovery_prefix"'})][0]
+        prefix = entities.Setting(id=hostname_prefix_id).read()
+        self.original_value = prefix.name
+        prefix.value = ""
+        prefix = prefix.update(['value'])
+        self.assertEqual(prefix.value, "")
+        prefix.value = self.original_value
+        prefix.update(['value'])
```

Comment:
I'm a bit confused by the last two lines, so we are setting the original_value to prefix *name* and after assert we are assigning it to prefix *value*. Also I'd use try - finally block to make sure the prefix is changed back also in case of unexpected failures e.g. when updating prefix

Reply:
@pondrejk this implementation is same as we do teardown() in tests. Please see similar implementation that we do in UI tests https://github.com/SatelliteQE/robottelo/blob/master/tests/foreman/ui/test_setting.py#L147 .

Reply:
@pondrejk good catch! updated the PR. Please revisit